Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -539,6 +574,204 @@ void GridTrackSizingAlgorithm::increaseSizesToAccommodateSpanningItems(const Gri
+            ASSERT(variant == TrackSizeComputationVariant::CrossingFlexibleTracks || itemSpan.integerSpan() > 1u);
+
+            filteredTracks.shrink(0);
+            growBeyondGrowthLimitsTracks.shrink(0);
+            LayoutUnit spanningTracksSize;
+            for (auto trackPosition : itemSpan) {
+                GridTrack& track = allTracks[trackPosition];
+                const auto& trackSize = track.cachedTrackSize();
+                spanningTracksSize += trackSizeForTrackSizeComputationPhase(phase, track, TrackSizeRestriction::ForbidInfinity);
+
+                if (!shouldProcessTrackForTrackSizeComputationPhase(phase, trackSize))
+                    continue;
+
+                filteredTracks.append(track);
+
+                if (trackShouldGrowBeyondGrowthLimitsForTrackSizeComputationPhase(phase, trackSize))
+                    growBeyondGrowthLimitsTracks.append(track);
+            }
+
+            if (filteredTracks.isEmpty())
+                continue;
+
+            spanningTracksSize += m_renderGrid->guttersSize(m_direction, itemSpan.startLine(), itemSpan.integerSpan(), availableSpace());
+
+            LayoutUnit extraSpace = masonryItemSizeForTrackSizeComputationPhase(phase, it->second.trackSize) - spanningTracksSize;
+            extraSpace = std::max<LayoutUnit>(extraSpace, 0);
+            auto& tracksToGrowBeyondGrowthLimits = growBeyondGrowthLimitsTracks.isEmpty() ? filteredTracks : growBeyondGrowthLimitsTracks;
+            distributeSpaceToTracks<variant, phase>(filteredTracks, &tracksToGrowBeyondGrowthLimits, extraSpace);
+        }
+
+        for (const auto& trackIndex : m_contentSizedTracksIndex) {
+            GridTrack& track = allTracks[trackIndex];
+            markAsInfinitelyGrowableForTrackSizeComputationPhase(phase, track);
+            updateTrackSizeForTrackSizeComputationPhase(phase, track);
+        }
+    };
+
+    StdSet<unsigned> spanLengths;
+    WTF::StdMultimap<unsigned, MasonryMinMaxTrackSizeWithGridSpan> multiTrackItems;
+    auto& allTracks = tracks(m_direction);
+
+    for (auto& gridTrackSpan : gridTrackSpans) {
+        for (auto trackIndex = 0u; trackIndex < allTracks.size(); trackIndex++) {
+            auto endLine = trackIndex + gridTrackSpan.first;
+            auto itemSpan = GridSpan::translatedDefiniteGridSpan(trackIndex, endLine);
+
+            if (itemSpan.integerSpan() == 1)
+                continue;
+
+            if (endLine > allTracks.size())
+                continue;
+
+            if (spanningItemCrossesFlexibleSizedTracks(itemSpan))
+                continue;
+
+            multiTrackItems.insert({ itemSpan.integerSpan(), MasonryMinMaxTrackSizeWithGridSpan { .trackSize = gridTrackSpan.second, .gridSpan = itemSpan } });
+
+            if (!spanLengths.contains(itemSpan.integerSpan()))
+                spanLengths.insert(itemSpan.integerSpan());
+        }
+    }
+
+    for (auto& multiTrackItem : definiteMultiTrackItems) {
```

Comment:
I don't think this should be `multiTrackItem` since we create a `multiTrackItems` vector above and this element is not from that vector

Reply:
This was re-written.